packages/node/src/utils/cosmos.ts

```diff
@@ -223,21 +223,21 @@ export async function fetchBlocksBatches(
   const blocks = await fetchCosmosBlocksArray(api, blockArray);
   return Promise.all(
     blocks.map(async (blockInfo) => {
-      const txHashes = blockInfo.txs;
-      if (txHashes === null || txHashes.length === 0) {
-        return <BlockContent>{
-          block: wrapBlock(blockInfo, []),
-          transactions: [],
-          messages: [],
-          events: [],
-        };
-      }
+      const txInfos =
+        blockInfo.txs.length <= 0
+          ? []
+          : await api.txInfoByHeight(blockInfo.header.height);
+
+      assert(
```

Comment:
I found this is an issue with the dictionary project https://github.com/subquery/juno-subql-dictionary

This is due to pruned nodes. I think there is a better way we can get the transaction info anyway.

Blocks already have the transactions in them, but we don't have the transaction results (logs, gas used). We can get them through [block_results](https://docs.tendermint.com/master/rpc/#/Info/block_results) query and that should give us all the data we need. 

There does appear to be a bug with this request. A fix is in the works https://github.com/cosmos/cosmjs/pull/1151.